compiler/src/dmd/opover.d

```diff
@@ -640,6 +622,45 @@ bool suggestBinaryOverloads(BinExp e, Scope* sc)
     return false;
 }
 
+/**
+ * If applicable, print an error relating to implementing / fixing `opOpAssign` or `opUnary` functions.
+ * Params:
+ *   exp = binary operation
+ *   sc = scope to try `opOpAssign!""` semantic in for error messages
+ *   parent = if `exp` was lowered from this `PreExp` or `PostExp`, mention `opUnary` as well
+ * Returns: `true` when an error related to `opOpAssign` was printed
+ */
+bool suggestOpOpAssign(BinAssignExp exp, Scope* sc, Expression parent)
+{
+    auto ad = isAggregate(exp.e1.type);
+    if (!ad)
+        return false;
+
+    if (parent && (parent.isPreExp() || parent.isPostExp()))
+    {
+        error(exp.loc, "operator `%s` not supported for `%s` of type `%s`", EXPtoString(parent.op).ptr, exp.e1.toChars(), ad.toChars());
+        errorSupplemental(ad.loc,
+            "perhaps implement `auto opUnary(string op : \"%s\")() {}`"~
+            " or `auto opOpAssign(string op : \"%s\")(int) {}`",
+            EXPtoString(stripAssignOp(parent.op)).ptr,
+            EXPtoString(stripAssignOp(exp.op)).ptr
+            );
+        return true;
+    }
+
+    if (const s = search_function(ad, Id.opOpAssign))
+    {
+        dotTemplateCall(exp.e1, Id.opOpAssign, opToArg(sc, exp.op), exp.e2).expressionSemantic(sc);
+    }
```

Comment:
function documentation says this function returns true if an error occurs. 
Is the call to `dotTemplateCall` or `expressionSemantic` expected to always error? 
If so a comment here would be good.

Reply:
Yes, I added a comment